metriq_gym/cli.py

```diff
@@ -24,19 +25,29 @@ def list_jobs(job_manager: JobManager) -> None:
     if not jobs:
         print("No jobs found.")
         return
-
+        
+    table_headers = LIST_JOBS_HEADERS.copy()
+    if show_index:
+        table_headers.insert(0, "Index")
+    
     table = [
         [
+            i,
+            job.id,
+            job.provider_name,
+            job.device_name,
+            job.job_type,
+            job.dispatch_time.strftime(TIMESTAMP_FORMAT),
+        ] if show_index else [
             job.id,
             job.provider_name,
             job.device_name,
             job.job_type,
             job.dispatch_time.strftime(TIMESTAMP_FORMAT),
         ]
-        for job in jobs
+        for i, job in enumerate(jobs)
     ]
-    print(tabulate(table, headers=LIST_JOBS_HEADERS, tablefmt="grid"))
-
+    print(tabulate(table, headers=table_headers, tablefmt="grid"))
```

Comment:
When I suggested to use `showindex`, I was talking about the actual `showindex` parameters of `tabulate`.